### ISSUE TITLE
`-Wodr` warnings when compiling minimp3 with LTO (regression from #78271)

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#78407]
### Godot version

4.1.beta (a83eb16fba5bb3da086b41cbd79e6f95b09eb8ee)

### System information

Windows & Linux, GCC

### Issue description

After merging #78271 I noticed new `-Wodr` warnings when building with GCC's LTO for official builds on Linux and Windows:

```
thirdparty/minimp3/minimp3_ex.h:88:3: warning: type 'struct mp3dec_ex_t' violates the C++ One Definition Rule [-Wodr]
   88 | } mp3dec_ex_t;
      |   ^
thirdparty/minimp3/minimp3_ex.h:88:3: note: a different type is defined in another translation unit
   88 | } mp3dec_ex_t;
      |   ^
thirdparty/minimp3/minimp3_ex.h:82:19: note: the first difference of corresponding definitions is field 'buffer'
   82 |     mp3d_sample_t buffer[MINIMP3_MAX_SAMPLES_PER_FRAME];
      |                   ^
thirdparty/minimp3/minimp3_ex.h:82:19: note: a field of same name but different type is defined in another translation unit
   82 |     mp3d_sample_t buffer[MINIMP3_MAX_SAMPLES_PER_FRAME];
      |                   ^
thirdparty/minimp3/minimp3_ex.h:88:3: note: type 'mp3d_sample_t' should match type 'mp3d_sample_t'
   88 | } mp3dec_ex_t;
      |   ^
thirdparty/minimp3/minimp3_ex.h:111:8: warning: 'mp3dec_ex_read_frame' violates the C++ One Definition Rule [-Wodr]
  111 | size_t mp3dec_ex_read_frame(mp3dec_ex_t *dec, mp3d_sample_t **buf, mp3dec_frame_info_t *frame_info, size_t max_samples);
      |        ^
thirdparty/minimp3/minimp3_ex.h:858:8: note: 'mp3dec_ex_read_frame' was previously declared here
  858 | size_t mp3dec_ex_read_frame(mp3dec_ex_t *dec, mp3d_sample_t **buf, mp3dec_frame_info_t *frame_info, size_t max_samples)
      |        ^
thirdparty/minimp3/minimp3_ex.h:858:8: note: code may be misoptimized unless '-fno-strict-aliasing' is used
```

I'll have to look at this again, seems like the header guards don't work, or we're compiling the header and the .cpp with different conditionals.

### Steps to reproduce

- `scons production=yes` with GCC

### Minimal reproduction project

n/a

**Comment by Ciapas-Linux** [comment]
Hi
I am compiling editor 4.1 under linux debian , well, I noticed that recent builds hang, I discovered that the editor crashes when loading or importing resources from mp3 files

**Comment by akien-mga** [comment]
I can reproduce the crash too with 7188ed5f9140f6e2a38a24ac3364bba741b45135, and not with 4.1 beta 2, so it's very likely caused by my change. Somehow even gdb doesn't manage to get a stacktrace for this crash.

I'll revert #78271 as it wasn't tested properly and introduced bugs.

**Comment by akien-mga** [comment]
Fixed by c79183817df84faaa7a67a32c00c5a2bd46dc632.
